libevmasm/PeepholeOptimiser.cpp

```diff
@@ -428,6 +428,47 @@ struct TruthyAnd: SimplePeepholeOptimizerMethod<TruthyAnd>
 	}
 };
 
+// (a == true) -> a and (true == a) -> a
+struct TruthyBooleanComparison: SimplePeepholeOptimizerMethod<TruthyBooleanComparison>{
+	static bool applySimple(
+		AssemblyItem const& _pushOrDupN,
+		AssemblyItem const& _iszero,
+		AssemblyItem const& _iszero1,
```

Comment:
I think we already have a rule that turns this `PUSH 1 ISZERO ISZERO` into `PUSH 1` (not sure where but I see in the output that it does get replaced) so this should still work if you just expect `PUSH 1`. Same for `false` with `PUSH 0`. But please test that.